src/functional/iterator/embedded.py

```diff
@@ -485,35 +564,35 @@ def setter(indices, value):
         def getter(indices):
             return a[_tupsum(indices, offsets)]
 
-        return LocatedField(getter, axises, dtype=a.dtype, setter=setter, array=a.__array__)
+        return LocatedFieldImpl(getter, axes, dtype=a.dtype, setter=setter, array=a.__array__)
 
     return _maker
 
 
-def index_field(axis, dtype=float):
-    return LocatedField(lambda index: index[0], (axis,), dtype)
+def index_field(axis: Dimension, dtype=float) -> LocatedField:
+    return LocatedFieldImpl(lambda index: index[0], (axis,), dtype)
 
 
-def constant_field(value: typing.Any, dtype: type) -> LocatedField:
-    return LocatedField(lambda _: value, (), dtype)
+def constant_field(value: Any, dtype: type) -> LocatedField:
+    return LocatedFieldImpl(lambda _: value, (), dtype)
 
 
 @builtins.shift.register(EMBEDDED)
-def shift(*offsets):
+def shift(*offsets: Union[Offset, int]):
     def impl(it):
-        return it.shift(*offsets)
+        return it.shift(*list(o.value if isinstance(o, Offset) else o for o in offsets))
 
     return impl
 
 
 @dataclass
 class Column:
-    axis: CartesianAxis
+    axis: str
     range: range  # noqa: A003
```

Comment:
Can `range` type hint be changed to `Iterable[int]` here?

Reply:
I don't think it would be correct. It should be a contiguous sequence of ints.